x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/TransportGetApiKeyAction.java

```diff
@@ -50,7 +50,11 @@ protected void doExecute(Task task, GetApiKeyRequest request, ActionListener<Get
             assert realm == null;
             // restrict username and realm to current authenticated user.
             username = authentication.getUser().principal();
-            realm = authentication.getAuthenticatedBy().getName();
+            if (authentication.getAuthenticatedBy().getType().equals(ApiKeyService.API_KEY_REALM_TYPE)) {
+                realm = (String) authentication.getMetadata().get(ApiKeyService.API_KEY_CREATOR_REALM);
+            } else {
+                realm = authentication.getAuthenticatedBy().getName();
+            }
```

Comment:
Should this logic be a function that is used in `ManageOwnApiKeyClusterPrivilege`?

Reply:
Sorry, the question I am raising is that you have here a way of extracting the realm name which is different than the way of extracting the realm name in `ManageOwnApiKeyClusterPrivilege`. I believe they should be the same.
It works as is, but it's brittle.

Reply:
The same logic is in 2 transport actions. That seems like a strong enough argument to move it to a static method on the server.